sdk/include/opentelemetry/sdk/metrics/view/attributes_processor.h

```diff
@@ -11,13 +11,25 @@ namespace metrics
 {
 using MetricAttributes = opentelemetry::sdk::common::AttributeMap;
 
+/**
+ * The AttributesProcessor is responsible for customizing which
```

Comment:
I wonder if this can be made internal/private to the SDK for now.

The specification only allows very limited manipulation on attributes (e.g. remove certain attributes from the measurement), this generic solution could work but it comes with perf cost (e.g. the virtual `AttributesProcessor::process` call).

Doesn't have to be a blocker though.

Reply:
Yes, I do suspect some perf cost associated with virtual and attributes copy. I have added a benchmark test to get cost statistics and would be helpful if have to consider performance improvement.

Also, I realized that we need to calculate the hash on the attribute map to store metrics data. Have added functionality to calculate the hash based on `std::hash` and boost::hash_combine`. As hash should be the same irrespective of the order of key/values in the AttributeMap, I have changed the internal storage for AttributeMap from unordered_map to map. This way we can avoid sorting of keys for every measurement.

Let me see how can we make this class `internal/private` to SDK, as C++ doesn't have a direct way of doing it. We can make `AttributeProcessor::process()` as private, and other SDK classes as its friend to it but would like to see a better approach if possible.